torch/_inductor/compile_fx.py

```diff
@@ -648,30 +648,41 @@ def codegen_and_compile(
             and (config.fx_graph_cache or fx_graph_remote_cache)
             and not aot_mode
         )
+        local = config.fx_graph_cache
+        remote = fx_graph_remote_cache
         set_feature_use("pytorch/remote_cache:fx_graph_memcache_version", use_cache)
 
-        if (
-            not config.force_disable_caches
-            and (config.fx_graph_cache or fx_graph_remote_cache)
-            and not aot_mode
-        ):
-            for i, input in enumerate(example_inputs):
-                if (
-                    isinstance(input, torch.Tensor)
-                    and input.device.type == "cuda"
-                    and i in static_input_idxs
-                ):
-                    input._is_inductor_static = True  # type: ignore[attr-defined]
+        # TODO: This is a hack purely to get some info to extract_tensor_metadata_for_cache_key,
+        # figure out how to not have to modify example inputs
+        for i, input in enumerate(example_inputs):
+            if (
+                isinstance(input, torch.Tensor)
+                and input.device.type == "cuda"
+                and i in static_input_idxs
+            ):
+                input._is_inductor_static = True  # type: ignore[attr-defined]
 
-            local = config.fx_graph_cache
-            remote = fx_graph_remote_cache
+        # TODO: Remove this short circuit once types are unified here
```

Comment:
Ok wait, isn't the entire point of this refactor to unify this type? Or is there another PR incoming to do that?

Reply:
This gets eliminated in https://github.com/pytorch/pytorch/pull/141695